pkg/k8s/controller.go

```diff
@@ -248,11 +248,35 @@ func (c *Controller) entityHandler(portEntity port.Entity, action EventActionTyp
 		}
 		klog.V(0).Infof("Successfully upserted entity '%s' of blueprint '%s'", portEntity.Identifier, portEntity.Blueprint)
 	case DeleteAction:
-		err := c.portClient.DeleteEntity(context.Background(), portEntity.Identifier, portEntity.Blueprint, c.portClient.DeleteDependents)
+		portEntities, err := c.portClient.SearchEntities(context.Background(), port.SearchBody{
+			Rules: []port.Rule{
+				{
+					Property: "$datasource",
+					Operator: "contains",
+					Value:    "port-k8s-exporter",
+				},
```

Comment:
You need also to add a rule to check that this specific exporter instance is the owner (checking the stateKey), like here: https://github.com/port-labs/port-k8s-exporter/blob/d726b09fc8d743f2ae1825c6eee509f99309a586/pkg/port/cli/entity.go#L100